test/core/end2end/generate_tests.bzl

```diff
@@ -401,12 +410,25 @@ def grpc_end2end_tests():
                 "//:grpc",
                 "//:gpr",
             ],
-            tags = ["no_windows"],
+            tags = _platform_support_tags(fopt),
         )
+
         for t, topt in END2END_TESTS.items():
             #print(_compatible(fopt, topt), f, t, fopt, topt)
             if not _compatible(fopt, topt):
                 continue
+
+            native.sh_test(
```

Comment:
It looks like this just runs the test without a poller specified. What is the behavior in this situation?

Reply:
It just runs the test with the default poller.

If poller is specified, it's set into the `GRPC_POLL_STRATEGY` env variable, otherwise default poller will be used (default grpc behavior).
https://github.com/grpc/grpc/blob/66c372b8ec1ec3c53e11ccdc47c394a4bb810cc2/test/core/end2end/end2end_test.sh#L20

Reply:
If that's the case, shouldn't that mean there's a test duplicated between this `sh_test` and the following loop which generates a test for each poller?

Reply:
yes, on linux it's basically "duplicated" in the sense there's one test with default poller and then a set of poller-specific tests.
This is mostly harmless because even the "default" poller test is a legitimate on linux, so if it gets run "by accident" it's fine (we're just wasting some time). The "default poller" test actually gets marked with "no_linux" tag and we're filtering it out in the bazel RBE .rc files (see diff) to make sure we don't waste time running the same test twice on our CI.

Ideally there would be a good way how to mark platform support for tests without needing to use --filter_build_tags and --filter_test_tags, but that's an open issue for now bazelbuild/bazel#3780 so we're doing what we can.